eng/common/scripts/artifact-metadata-parsing.ps1

```diff
@@ -0,0 +1,471 @@
+. (Join-Path $PSScriptRoot SemVer.ps1)
+
+$VERSION_REGEX = "(?<major>\d+)(\.(?<minor>\d+))?(\.(?<patch>\d+))?((?<pre>[^0-9][^\s]+))?"
+$SDIST_PACKAGE_REGEX = "^(?<package>.*)\-(?<versionstring>$VERSION_REGEX$)"
+
+# Posts a github release for each item of the pkgList variable. SilentlyContinue
+function CreateReleases($pkgList, $releaseApiUrl, $releaseSha) {
+  foreach ($pkgInfo in $pkgList) {
+    Write-Host "Creating release $($pkgInfo.Tag)"
+
+    $releaseNotes = ""
+    if ($pkgInfo.ReleaseNotes[$pkgInfo.PackageVersion].ReleaseContent -ne $null) {
+      $releaseNotes = $pkgInfo.ReleaseNotes[$pkgInfo.PackageVersion].ReleaseContent
+    }
+
+    $isPrerelease = $False
+    if ($pkgInfo.PackageVersion -match $VERSION_REGEX) {
+      $preReleaseLabel = $matches["pre"]
+      $isPrerelease = ![string]::IsNullOrEmpty($preReleaseLabel)
+    }
+
+    $url = $releaseApiUrl
+    $body = ConvertTo-Json @{
+      tag_name         = $pkgInfo.Tag
+      target_commitish = $releaseSha
+      name             = $pkgInfo.Tag
+      draft            = $False
+      prerelease       = $isPrerelease
+      body             = $releaseNotes
+    }
+
+    $headers = @{
+      "Content-Type"  = "application/json"
+      "Authorization" = "token $($env:GH_TOKEN)"
+    }
+
+    Invoke-WebRequest-WithHandling -url $url -body $body -headers $headers -method "Post"
+  }
+}
+
+function Invoke-WebRequest-WithHandling($url, $method, $body = $null, $headers = $null) {
+  $attempts = 1
+
+  while ($attempts -le 3) {
+    try {
+      return Invoke-RestMethod -Method $method -Uri $url -Body $body -Headers $headers
```

Comment:
Is there a reason not to use the MaxRetryCount option natively on Invoke-RestMethod?

Reply:
I knew about MaxRetryCount. At the time I'm pretty certain we wanted explicit output for the reset information.